bgpd/bgp_fsm.c

```diff
@@ -1892,6 +1892,22 @@ static enum bgp_fsm_state_progress bgp_start(struct peer_connection *connection)
 				 __func__, peer->connection->fd);
 			return BGP_FSM_FAILURE;
 		}
+		/* after connect is called(), getpeername is able to return
+		 * port and address on non established streams
+		 */
+		bgp_getsockname(peer);
```

Comment:
Wouldn't be better to move this logic to `bgp_connect_check()`?

Reply:
moving to bgp_connect_check() will delay the time when context is updated. as the client can perform a "show", or "snmpwalk" during that period, the value can be 0, and it is not desirable to have that 0 value.

What I can do however, is to encapsulate this logic in a function called `bgp_connect_in_progress_update_connection()`